realm/src/main/java/io/realm/Realm.java

```diff
@@ -656,6 +687,15 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (cachedDurability.value != newDurability.value) {
+                throw new IllegalArgumentException("Configurations cannot have different durabilities for the " +
+                        "same realm filename/identifier: " + newConfiguration.getRealmFileName() + " " +
```

Comment:
"in-memory" is already a concept from the SQLite API's, so people are familiar with it. It is of course a bit problematic that we actually write to disk, but at least "in-memory" is a term people would be familiar with on Android.